daml/Marketplace/Trading/Service.daml

```diff
@@ -25,8 +31,46 @@ template Service
     key (operator, provider, customer) : (Party, Party, Party)
     maintainer key._1
 
+    let
+      withdrawFromAllocationAccount : Order.TradeCollateral -> Update Order.TradeCollateral
+      withdrawFromAllocationAccount = \tc -> case tc of
+        (Order.Cleared party)         -> return $ Order.Cleared party
+        (Order.Collateral depositCid) -> Order.Collateral <$> exerciseByKey @AllocationAccountRule.T allocationAccount.id AllocationAccountRule.Withdraw with transferTo = tradingAccount, ..
+
     controller customer can
-      nonconsuming RequestCreateOrder : (ContractId Order.T, ContractId CreateOrderRequest, ContractId AssetDeposit)
+      nonconsuming RequestCreateClearedOrder : Either (ContractId Order.T)
```

Comment:
Doesn't feel right to me that we have a second function for creating cleared order. As per previous comment, I think we need to separate the concept of an order from collateral/clearing. Then have a single `RequestCreateOrder` with collateral/clearing as additional arguments, separate from the order economics.

Reply:
Makes sense - I originally had it that way and split it out to be a bit cleaner before I had the `TradingCollateral` type, so I can put it back into one function.